Libplanet.Headless/ReducedStore.cs

```diff
@@ -0,0 +1,153 @@
+using Libplanet.Store;
+using Libplanet.Tx;
+
+namespace Libplanet.Headless
+{
+    /// <summary>
+    /// A <see cref="IStore"> decorator that reduce space consumption by omitting input calls which
+    /// are unused by Nine Chronicles.
+    /// <para>Calls on this will be forwarded to its <see cref="InternalStore"/>, except for:</para>
+    /// <list type="bullet">
+    /// <item><description><see cref="PutTxExecution(TxSuccess)"/></description></item>
+    /// </list>
+    /// </summary>
+    public sealed class ReducedStore : IStore
+    {
+        public ReducedStore(IStore internalStore)
+        {
+            InternalStore = internalStore;
+        }
+
+        public IStore InternalStore { get; }
+
+        public long AppendIndex(Guid chainId, BlockHash hash) =>
+            InternalStore.AppendIndex(chainId, hash);
+
+        public bool ContainsBlock(BlockHash blockHash) =>
+            InternalStore.ContainsBlock(blockHash);
+
+        public bool ContainsTransaction(TxId txId) =>
+            InternalStore.ContainsTransaction(txId);
+
+        public long CountBlocks() =>
+            InternalStore.CountBlocks();
+
+        public long CountIndex(Guid chainId) =>
+            InternalStore.CountIndex(chainId);
+
+        public long CountTransactions() =>
+            InternalStore.CountTransactions();
+
+        public bool DeleteBlock(BlockHash blockHash) =>
+            InternalStore.DeleteBlock(blockHash);
+
+        public void DeleteChainId(Guid chainId) =>
+            InternalStore.DeleteChainId(chainId);
+
+        public bool DeleteTransaction(TxId txid) =>
+            InternalStore.DeleteTransaction(txid);
+
+        public void ForkBlockIndexes(
+            Guid sourceChainId,
+            Guid destinationChainId,
+            BlockHash branchpoint
+        ) =>
+            InternalStore.ForkBlockIndexes(sourceChainId, destinationChainId, branchpoint);
+
+        public void ForkTxNonces(Guid sourceChainId, Guid destinationChainId) =>
+            InternalStore.ForkTxNonces(sourceChainId, destinationChainId);
+
+        public Block<T> GetBlock<T>(BlockHash blockHash)
+            where T : IAction, new() =>
+            InternalStore.GetBlock<T>(blockHash);
+
+        public BlockDigest? GetBlockDigest(BlockHash blockHash) =>
+            InternalStore.GetBlockDigest(blockHash);
+
+        public long? GetBlockIndex(BlockHash blockHash) =>
+            InternalStore.GetBlockIndex(blockHash);
+
+        public DateTimeOffset? GetBlockPerceivedTime(BlockHash blockHash) =>
+            InternalStore.GetBlockPerceivedTime(blockHash);
+
+        public Guid? GetCanonicalChainId() =>
+            InternalStore.GetCanonicalChainId();
+
+        public Transaction<T> GetTransaction<T>(TxId txid) where T : IAction, new() =>
+            InternalStore.GetTransaction<T>(txid);
+
+        public TxExecution GetTxExecution(BlockHash blockHash, TxId txid) =>
+            InternalStore.GetTxExecution(blockHash, txid);
```

Comment:
How about throwing `NotSupportedException`?

Reply:
We still need `TxFailure`, even though `TxSuccess` is unused.

Reply:
🆗